src/dataspec/factories.py

```diff
@@ -1171,6 +1172,142 @@ def str_contains_phonenumber(s: str) -> Iterator[ErrorDetails]:
         )
 
 
+def rename_spec(  # noqa: MC0001  # pylint: disable=too-many-arguments
+    tag: Optional[Tag] = None,
+    replacements: Optional[Mapping[Hashable, Union[Hashable, List[Hashable]]]] = None,
+    retain_replaced_keys: bool = False,
```

Comment:
I know it's a big change in a different direction but what if you split this into two specs, rename and copy_value specs. They could reuse this guy under the hood but it might make the api a bit simpler.

Reply:
How would the copy_value spec work?

Reply:
copy_value_spec would be rename_spec but hwere retain_replaced_keys = True